src/main/java/org/opensearch/neuralsearch/processor/ScoreNormalizer.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.processor;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.Optional;
+
+import lombok.extern.log4j.Log4j2;
+
+import org.apache.lucene.search.ScoreDoc;
+import org.apache.lucene.search.TopDocs;
+import org.opensearch.neuralsearch.search.CompoundTopDocs;
+
+/**
+ * Abstracts normalization of scores in query search results.
+ */
+@Log4j2
+public class ScoreNormalizer {
+
+    /**
+     * Performs score normalization based on input combination technique. Mutates input object by updating normalized scores.
+     * @param queryTopDocs original query results from multiple shards and multiple sub-queries
+     * @param normalizationTechnique exact normalization method that should be applied
+     */
+    public void normalizeScores(final CompoundTopDocs[] queryTopDocs, final ScoreNormalizationTechnique normalizationTechnique) {
```

Comment:
This can be a static function, as we are not storing any states here.

Reply:
ack